### ISSUE TITLE
Detected as virus

### ISSUE THREAD

**Issue by Nooby-001** [psoukie/zipchord#131]
VirusTotal says: "12 security vendors and no sandboxes flagged this file as malicious" and "Gen:Variant.Tedy.269682"

Link: https://www.virustotal.com/gui/file/7c7753261eb733734d397c9f7c9b9091c03b4b23dd4294d674704c2f8da43c6d/detection

Past versions have had two or three detections on VirusTotal, my PC antivirus, Avast, never detected anything but whenever I ran a new version it sent it to "their labs" with no problems to report.

**Comment by psoukie** [comment]
I'll keep an eye on how I could improve this. Sadly, AutoHotkey -- as any scripting language -- can be used maliciously. I am not sure how these programs decide whether the program might be harmful.

Avast is right. There is absolutely no code that modifies or deletes any files on your computer (other than those created by ZipChord), it does not send or receive any information over the internet whatsoever, and it does not store your typing even in memory beyond what's immediately entered.

I will add digital signatures to releases on GitHub which might reduce some of the false positives. I will also keep all unnecessary features away from the compiled executable file. (So this will keep the debugging and console out of the releases, which could also help -- right now, the debugging is included in the exe, but only runs on demand, and only saves the file locally. Still this change could help.)

If anyone has ideas and recommendations how to reduce the false positives, please let me know.

**Comment by zokizuan** [comment]
https://www.virustotal.com/gui/file/7c7753261eb733734d397c9f7c9b9091c03b4b23dd4294d674704c2f8da43c6d/detection

**Comment by Nooby-001** [comment]
I was searching for "Gen:Variant.Tedy.269682" one of the VirusTotal results, and found this, don't know if it applies to AutoHotkey but it might help:

"I'd like also to mention that If i changed the assembly name the antivirus is no longer consider the new file virus, but it looks that it considers it virus because I invoke it many times, with different parameters"

and

"to fix this issue simply add icon to the program, im not even joking, it works."

Link:
https://stackoverflow.com/questions/58010466/bitdefender-detects-my-console-application-as-genvariant-ursu-56053

**Comment by psoukie** [comment]
For 2.0.1, I even removed the debugging feature in an effort to reduce the false positives from these security vendors, but it did not help. (The icon thing shows how random it is, but ZipChord has had an icon for a long time.)

My advice is to use good antivirus and antimalware programs. If any of the more mainstream and established providers (AVG, Avast, Google, Malwarebytes, Kaspersky etc.) flag ZipChord, please let me know.

**Comment by Nooby-001** [comment]
Yeah, the latest Zipchord gets 13 flags at VirusTotal. I put more info in my latest post.

I'll close this issue.

**Comment by psoukie** [comment]
Just a note that I made a quick test that shows that is not that ZipChord is now doing more things, but rather, some antivirus companies are adding detection for new viruses that flags more software. 

I compiled version 1.2 and only changed one text placeholder from "999 chords" to "99 chords" (so that the antivirus would not just retrieve a previous result, but had to analyze it as a "new" file.)

The result is that 13 vendors are now (incorrectly) flagging even ZipChord 1.2 as malicious. ZipChord 2.1 should be even easier to correctly diagnose because it does not do things that could be confused for being malicious. (Like ability to create and record a log file.)
